src/view/components/calculation-history/CalculationHistory.tsx

```diff
@@ -13,36 +14,65 @@ type CountryDropdownType = {
 export const CalculationHistory = ({
     calculationHistory,
 }: CountryDropdownType) => {
+    const [isListClosed, setIsListClosed] = React.useState(true);
     return (
-        <div>
-            <ul>
-                {calculationHistory.map((calculation, index) => (
-                    <li key={index}>
-                        <p>Bytes: {formatBytes(calculation.bytes)}</p>
-                        <p>
-                            Emissions: {formatEmissions(calculation.emissions)}
-                        </p>
-                        <p>
-                            Specific Emissions:{" "}
-                            {formatEmissions(calculation.specificEmissions)}
-                        </p>
-                        <p>
-                            Date & Time:{" "}
-                            {unixTimeToDateString(calculation.unixTimeMs)}
-                        </p>
-                        <p>User: {calculation.userType}</p>
-                        <ul>
-                            {Array.from(calculation.selectedCountries).map(
-                                ([countryName, percentage]) => (
-                                    <li key={countryName}>
-                                        {countryName}: {percentage}
-                                    </li>
-                                )
-                            )}
-                        </ul>
-                    </li>
-                ))}
-            </ul>
-        </div>
+        <>
+            {isListClosed ? (
+                <Button
+                    colour="light-green"
+                    type="text"
+                    onClick={() => setIsListClosed(false)}
+                >
+                    View History
+                </Button>
+            ) : (
+                <ul className="overflow-y-scroll h-96 rounded-2xl shadow bg-nyanza p-4">
+                    <h2 className="font-semibold text-center mb-2 text-lg">
+                        Calculation History
+                    </h2>
+                    {calculationHistory.map((calculation, index) => {
+                        const [day, time] = unixTimeToDateString(
```

Comment:
Have you considered `new Date(calculation.unixTimeMs)` instead of string.split? 🤔